test/libsolidity/util/TestFileParser.h

```diff
@@ -0,0 +1,338 @@
+	T(RBrack, "]", 0)              \
+	T(LBrace, "{", 0)              \
+	T(RBrace, "}", 0)              \
+	T(Sub,    "-", 0)              \
+	T(Colon,  ":", 0)              \
+	T(Comma,  ",", 0)              \
+	T(Period, ".", 0)              \
+	T(Arrow, "->", 0)              \
+	T(Newline, "//", 0)            \
+	/* Literals & identifier */    \
+	T(Comment, "#", 0)             \
+	T(Number, "number", 0)         \
+	T(Identifier, "identifier", 0) \
+	/* type keywords */            \
+	K(Ether, "ether", 0)           \
+	/* special keywords */         \
+	K(Failure, "FAILURE", 0)       \
+
+enum class SoltToken : unsigned int {
+#define T(name, string, precedence) name,
+	SOLT_TOKEN_LIST(T, T)
+	NUM_TOKENS
+#undef T
+};
+
+/**
+ * The purpose of the ABI type is the storage of type information
+ * retrieved while parsing a test. This information is used
+ * for the conversion of human-readable function arguments and
+ * return values to `bytes` and vice-versa.
+ * Defaults to None, a 0-byte representation. 0-bytes
+ * can also be interpreted as Failure, which means
+ * either a REVERT or another EVM failure.
+ */
+struct ABIType
+{
+	enum Type {
+		UnsignedDec,
+		SignedDec,
+		Failure,
+		None
+	};
+	Type type = ABIType::None;
+	size_t size = 0;
+};
+
+/**
+ * Helper that can hold format information retrieved
+ * while scanning through a parameter list in sol_t.
+ */
+struct FormatInfo
+{
+	bool newline;
+};
+
+/**
+ * Parameter abstraction used for the encoding and decoding of
+ * function parameter and expectation / return value lists.
+ * A parameter list is usually a comma-separated list of literals.
+ * It should not be possible to call create a parameter holding
+ * an identifier, but if so, the ABI type would be invalid.
+ */
+struct Parameter
+{
+	/// ABI encoded / decoded `bytes` of values.
+	/// These `bytes` are used to pass values to function calls
+	/// and also to store expected return vales. These are
+	/// compared to the actual result of a function call
+	/// and used for validating it.
+	bytes rawBytes;
+	/// Types that were used to encode `rawBytes`. Expectations
+	/// are usually comma separated literals. Their type is auto-
+	/// detected and retained in order to format them later on.
+	ABIType abiType;
+	/// Format info attached to the parameter. It handles newlines given
+	/// in the declaration of it.
+	FormatInfo format;
+};
+using ParameterList = std::vector<Parameter>;
+
+/**
+ * Represents the expected result of a function call after it has been executed. This may be a single
+ * return value or a comma-separated list of return values. It also contains the detected input
+ * formats used to convert the values to `bytes` needed for the comparison with the actual result
+ * of a call. In addition to that, it also stores the expected transaction status.
+ * An optional comment can be assigned.
+ */
+struct FunctionCallExpectations
+{
+	/// Representation of the comma-separated (or empty) list of expectated result values
```

Comment:
```suggestion
	/// Representation of the comma-separated (or empty) list of expected result values
```

Reply:
Actually I wonder why that wasn't caught by the spell checker :-).

Reply:
Maybe because it's commented "inline"? This needs to fail though ;)